src/audio/igo_nr/igo_nr.c

```diff
@@ -697,11 +749,13 @@ static int32_t igo_nr_prepare(struct comp_dev *dev)
 	/* Clear in/out buffers */
 	memset(cd->in, 0, IGO_NR_IN_BUF_LENGTH * sizeof(int16_t));
 	memset(cd->out, 0, IGO_NR_OUT_BUF_LENGTH * sizeof(int16_t));
-	cd->in_wpt = 0;
-	cd->out_rpt = 0;
+
+	/* Check active_channel_idx validity */
+	if (cd->config.active_channel_idx >= SOF_IPC_MAX_CHANNELS)
+		cd->config.active_channel_idx = 0;
```

Comment:
the better thing to do is check on the cmd step, and if invalid, log an error and return -EINVAL, what you have here leads to silent and delayed failure possibly

Reply:
Okay. Will reject the channel mismatch in early stage.